src/index.ts

```diff
@@ -285,4 +289,20 @@ export class JolocomSDK {
     })
     await this.storage.store.identity(id)
   }
+
+  public async deleteIdentityData(did: string): Promise<void> {
+    let identity = await this.resolve(did)
+    await this.storage.delete.encryptedWallet(did)
+    await this.storage.delete.verifiableCredentials([
+      { subject: did },
+      { issuer: did },
+    ])
+    await this.storage.delete.identity(did)
+    let query: InteractionQueryAttrs[] = []
+    identity.publicKeySection.forEach((pk) => {
+      let keyId = `${did}${pk.id}`
+      query.push({ initiator: keyId }, { responder: keyId })
+    })
+    await this.storage.delete.interactions(query)
```

Comment:
In what case would we use different public keys for different interactions?

Reply:
we don't currently have other keys, but just to support future possibilities